### ISSUE TITLE
Wrong routes rendering in relation to clusterDomain

### ISSUE THREAD

**Issue by apolovov** [istio/istio#35676]
### Bug Description

In some cases, istio cuts off first-level domain in routes domains. Reproducing:

* Create cluster with `clusterDomain: aaa.bbb.not-example.wtf`.
* In namespace `test`, enable istio and deploy the sleep app to it.
* Create two `ServiceEntry`-es:
```
---
kind: ServiceEntry
apiVersion: networking.istio.io/v1alpha3
metadata:
  name: aaa
  namespace: test
spec:
  endpoints:
    - address: aaa.example.wtf
      ports:
        http-7777: 7777
  exportTo:
    - .
  hosts:
    - aaa.example.wtf
  location: MESH_EXTERNAL
  ports:
    - name: http-7777
      number: 7777
      protocol: HTTP
  resolution: DNS
---
kind: ServiceEntry
apiVersion: networking.istio.io/v1alpha3
metadata:
  name: bbb
  namespace: test
spec:
  endpoints:
    - address: bbb.example.net
      ports:
        http-8888: 8888
  exportTo:
    - .
  hosts:
    - bbb.example.net
  location: MESH_EXTERNAL
  ports:
    - name: http-8888
      number: 8888
      protocol: HTTP
  resolution: DNS
```
* Check the routes for port 8888 (everything is fine and domains are as expected):
```
# istioctl -n test pc routes sleep-557747455f-95ff6 --name 8888 -o yaml
- name: "8888"
  validateClusters: false
  virtualHosts:
  - domains:
    - bbb.example.net
    - bbb.example.net:8888
    includeRequestAttemptCount: true
    name: bbb.example.net:8888
    routes:
...
```
* Check the routes for port 7777 (the domains are inconsistent, first-level "wtf" is missing):
```
# istioctl -n test pc routes sleep-557747455f-95ff6 --name 7777 -o yaml
- name: "7777"
  validateClusters: false
  virtualHosts:
  - domains:
    - aaa.example
    - aaa.example:7777
    includeRequestAttemptCount: true
    name: aaa.example.wtf:7777
    routes:
...
```

### Version

```prose
# istioctl version
client version: 1.11.2
control plane version: 1.10.1
data plane version: 1.10.1 (6 proxies)

# kubectl version
Client Version: version.Info{Major:"1", Minor:"21", GitVersion:"v1.21.3", GitCommit:"ca643a4d1f7bfe34773c74f79527be4afd95bf39", GitTreeState:"clean", BuildDate:"2021-07-15T21:04:39Z", GoVersion:"go1.16.6", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"21", GitVersion:"v1.21.3", GitCommit:"0000000000000000000000000000000000000000", GitTreeState:"archive", BuildDate:"2021-07-21T15:25:26Z", GoVersion:"go1.16.3", Compiler:"gc", Platform:"linux/amd64"}
```

**Comment by howardjohn** [comment]
Tried to reproduce in unit tests:
```diff
diff --git a/pilot/pkg/networking/core/v1alpha3/httproute_test.go b/pilot/pkg/networking/core/v1alpha3/httproute_test.go
index 83cdee0d66..b00fe82f76 100644
--- a/pilot/pkg/networking/core/v1alpha3/httproute_test.go
+++ b/pilot/pkg/networking/core/v1alpha3/httproute_test.go
@@ -129,6 +129,18 @@ func TestGenerateVirtualHostDomains(t *testing.T) {
                        },
                        want: []string{"google.local", "google.local:8123"},
                },
+               {
+                       name: "35676",
+                       service: &model.Service{
+                               Hostname:     "aaa.example.wtf",
+                               MeshExternal: true,
+                       },
+                       port: 7777,
+                       node: &model.Proxy{
+                               DNSDomain: "tests.svc.aaa.bbb.not-example.wtf",
+                       },
+                       want: []string{},
+               },
                {
                        name: "ipv4 domain",
                        service: &model.Service{
```

Got similar but not the same results: `aaa.example aaa.example.wtf aaa.example.wtf aaa.example.wtf:7777 aaa.example.wtf:7777 aaa.example:7777`. Definitely looks like a bug since there is a duplicate there, at the very least.

**Comment by howardjohn** [comment]
https://github.com/istio/istio/commit/89553fedcd256afda5b312316a522a6f1f99adc4 is likely _related_. Not sure it was actually working before then though. @ramaraochavali

**Comment by fams** [comment]
We're experiencing this issue in istio 1.11.7. Is it will be fixed in 1.11.x or 1.12.x? I saw a fix in 1.13.0 release notes only
